src/main/java/HttpServerRunner.java

```diff
@@ -1,44 +1,33 @@
 package com.td.HttpServer;
 import java.io.IOException;
+import java.net.ServerSocket;
 
 public class HttpServerRunner {
+  private ServerSocket serverSocket;
   private HandlerRouter handlerRouter;
   private IRequestBuilder httpRequestBuilder;
   private HttpResponseWriter httpResponseWriter;
 
-  public HttpServerRunner(HandlerRouter handlerRouter, IRequestBuilder httpRequestBuilder, HttpResponseWriter httpResponseWriter) {
+  public HttpServerRunner(ServerSocket serverSocket, HandlerRouter handlerRouter, IRequestBuilder httpRequestBuilder, HttpResponseWriter httpResponseWriter) {
+    this.serverSocket = serverSocket;
     this.handlerRouter = handlerRouter; 
     this.httpRequestBuilder = httpRequestBuilder;
     this.httpResponseWriter = httpResponseWriter;
   }
 
-  public void run(ClientSocketIO client) {
-    try {
-      Ihandler handler = getHandler(client);
-      sendResponse(client, handler);
+  public void run() {
+    while (true) {
+      ClientSocketIO client = new ClientSocketIO(serverSocket);
+      try {
+        client.openClientConnection();
+      }
+      catch (BadConnectionException e) {
+        e.printStackTrace();
+        continue;
+      }
+      HttpServerThread httpServerThread = new HttpServerThread(client, handlerRouter, httpRequestBuilder, httpResponseWriter);
+      Thread thread = new Thread(httpServerThread);
```

Comment:
Similar idea here. `Thread` is an API class. `ClientSocketIO`, `HttpServerThread`, `ServerSocket`, and other several classes in here are all your code (creamy center).

Can you rip this API dependency out of here?

Taking it further - can you structure your hierarchy in such a way that "nobody knows" (and nobody cares) that this thing is actually multithreaded, _except_ for the thing that's actually responsible for the multithreading? Can you isolate the "mutlithreading" responsibility into one class?